src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,122 @@
 package core.basesyntax;
 
+import java.util.LinkedList;
+import java.util.List;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int INITIAL_CAPACITY = 16;
+    private static final float LOAD_CAPACITY = 0.75f;
+    private List<Node<K, V>>[] buckets;
+    private int size;
+
+    public MyHashMap() {
+        buckets = new List[INITIAL_CAPACITY];
+        size = 0;
+    }
 
     @Override
     public void put(K key, V value) {
-
+        int index = getIndex(key);
+        if (buckets[index] != null) {
+            Node<K, V> lastNode = null;
+            for (Node<K, V> node : buckets[index]) {
+                lastNode = node;
+                if (node.key == null && key == null) {
+                    node.value = value;
+                    return;
+                } else if (node.key == null || key == null) {
+                    continue;
+                } else if (node.key.equals(key)) {
+                    node.value = value;
+                    return;
+                }
+            }
+            Node<K, V> newNode = createNewNode(key, value);
+            buckets[index].add(newNode);
+            lastNode.next = newNode;
+        }
+        if (buckets[index] == null) {
+            buckets[index] = new LinkedList<>();
+            buckets[index].add(createNewNode(key, value));
+        }
+        size++;
+        resizeIfNeeded();
     }
 
     @Override
     public V getValue(K key) {
+        int index = getIndex(key);
+        if (buckets[index] != null) {
+            for (Node<K, V> node : buckets[index]) {
+                if (key == null && node.key == null) {
+                    return node.value;
+                }
+                if (node.key == null) {
+                    continue;
+                }
+                if (node.key.equals(key)) {
+                    return node.value;
+                }
+            }
```

Comment:
instead of many conditions and `continue`, which makes the code flow harder to follow, use `Objects.equals`